source/common/network/utility.cc

```diff
@@ -272,12 +273,17 @@ bool Utility::isLocalConnection(const Network::ConnectionSocket& socket) {
     }
 
     if (ifa->ifa_addr->sa_family == af_look_up) {
-      const auto* addr = reinterpret_cast<const struct sockaddr_storage*>(ifa->ifa_addr);
-      auto local_address = Address::addressFromSockAddr(
-          *addr, (af_look_up == AF_INET) ? sizeof(sockaddr_in) : sizeof(sockaddr_in6));
-
-      if (remote_address == local_address) {
-        return true;
+      if (af_look_up == AF_INET) {
```

Comment:
Could you move this comparison into `isSameIPAddress(const Address::Instance& address1, const struct sockaddr *address2)`?

Reply:
`isSameIPAddress()` already checks the IP version, but here, within a loop, we don't want to create temporary IP Address Instances so we are open-coding the comparison.

However, upon a second look I removed the assert in `isSameIPAddress()` so that it is safe to call on any type of an Address.

Reply:
Note that I'm suggesting extracting this to `isSameIPAddress()` with different arguments, so you'd invoke it using `isSameIPAddress(*remote_address, ifa->ifa_addr)` without creating temporary `Address::Instance` object (which I agree would be an overkill).

Basically, extracting this code for readability, since compiler is going to inline it here anyway.

I'm fine with you leaving it here, if feel strongly about it, though.

Reply:
Friendly ping on what you prefer to do here (leave it as-is, in which case it's ready to be merged, or extracting it to `isSameIPAddress(const Address::Instance&, const struct sockaddr *)`), to make sure that we're not stuck waiting on each other.